subprojects/parseq/src/test/java/com/linkedin/parseq/AbstractTaskTest.java

```diff
@@ -96,6 +97,34 @@ public void testRecover(int expectedNumberOfTasks) {
     runAndWait("AbstractTaskTest.testRecoverFailure", failure);
     assertEquals((int) failure.get(), -1);
     assertEquals(countTasks(failure.getTrace()), expectedNumberOfTasks);
+
+    Task<Integer> recoveredWithExceptionFilter = getFailureTask()
+            .map("strlen", String::length)
+            .recover("", RuntimeException.class, e -> -1);
+    runAndWait("AbstractTaskTest.testRecoverWithExceptionFilterSuccess", recoveredWithExceptionFilter);
+    assertEquals((int) recoveredWithExceptionFilter.get(), -1);
+    assertEquals(countTasks(recoveredWithExceptionFilter.getTrace()), expectedNumberOfTasks);
+
+    Task<Integer> notRecoveredWithExceptionFilter = getFailureTask()
+            .map("strlen", String::length)
+            .recover("", ArithmeticException.class, e -> -1);
+    runAndWaitException("AbstractTaskTest.testRecoverWithExceptionFilterFailure", notRecoveredWithExceptionFilter, RuntimeException.class);
+    assertTrue(notRecoveredWithExceptionFilter.isFailed());
+    assertEquals(countTasks(recoveredWithExceptionFilter.getTrace()), expectedNumberOfTasks);
+
+    Task<String> recoveredWithExceptionFilterAndSubclassesForward = getFailureTask() // Failed with RuntimeError
```

Comment:
This is good, but there are two scenarios that should also be tested:
1. Second recover is selected. e.g. `BarException` is thrown, first recover is `FooException` (unrelated exception), second exception is `BarException`.
2. Case where the first recover itself results in an exception, would it then be caught by the second recover?

Reply:
Fixed, thank you @evanw555 .